### PR TITLE
Report improvements

### LINKED ISSUE #317: Report should include project namespaces
It would be useful if the client names in the table in client conformance report included namespace : `sigstore/sigstore-python` instead of just `sigstore-python`. The name could be a link as well

### PR DESCRIPTION
**Pull request by jku**
Improvements to the client conformance report:
* include full project name and url in the report (e.g. [sigstore/sigstore-python](https://github.com/sigstore/sigstore-python))
* accept results that have failures (otherwise we have missing data when tests fail)
* show "User-managed keys" (#301) as a feature check in the report (it's not the most important feature but it's the only interesting one where we have variance among clients)

fixes #317

### CODE REVIEW

File: .github/scripts/generate_client_report.py [hunk 1]
```diff
@@ -51,6 +54,8 @@ def __init__(self, report_path: Path):
                 self.rekor2_verify = test["outcome"] == "passed"
             elif nodeid == "test/test_bundle.py::test_sign_verify_rekor2":
                 self.rekor2_sign = test["outcome"] == "passed"
+            elif nodeid == "test/test_bundle.py::test_verify[PATH-managed-key-happy-path]":
```
[the remark this answers is not in the document]
Comment (jku) [reply .github/scripts/generate_client_report.py]: right: I intended these checks as a service to people who want to see which features are supported by which clients. so the test has to pass (not fail, not xfail) for the check to appear.

Now a client could support a feature but just have a bug in the implementation... that feels like an edge case the client devs can easily handle by fixing the bug.

**Comment by jku** [comment]
Report looks ok: https://sigstore.github.io/sigstore-conformance/.  at time of writing sigstore-python is the only client with updated results.

the only issue seems to be that the link for clients that have not upgraded yet is empty (so points to https://sigstore.github.io/sigstore-conformance/ instead of the client project): I can live with that
